src/socket/socket_client_ws.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+const W3CWebSocket = require('websocket').w3cwebsocket
+
+class WebSocketClient {
+  constructor(self, host, port, protocol, options) {
+    this.self = self
+    this.host = host
+    this.port = port
+    this.protocol = protocol
+    this.options = options
+    this.client = null
+  }
+
+  async connect() {
+    const url = `${this.protocol}://${this.host}:${this.port}`
+
+    const client = new W3CWebSocket(
```

Comment:
Placeholder doc link for now https://github.com/theturtle32/WebSocket-Node/blob/master/docs/W3CWebSocket.md#constructor